cmds/core/sync/sync.go

```diff
@@ -1,12 +1,63 @@
-// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Copyright 2016-2020 the u-root Authors. All rights reserved
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
 // sync command in Go.
+//
+// Synopsis:
+//		sync [-df] [FILE]
+//
+
 package main
 
-import "syscall"
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+var (
+	data       = flag.Bool("data", false, "sync file data, no metadata")
+	filesystem = flag.Bool("filesystem", false, "commit filesystem caches to disk")
+)
+
+func init() {
+	flag.BoolVar(data, "d", false, "")
+	flag.BoolVar(filesystem, "f", false, "")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION] [FILE]...\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+}
+
+func doSyscall(callType uintptr) {
+	for _, fileName := range flag.Args() {
+		f, err := os.OpenFile(fileName, syscall.O_RDONLY|syscall.O_NOCTTY|syscall.O_CLOEXEC, 0644)
+		if err != nil {
+			log.Fatal(err)
+		}
+		_, _, err = syscall.Syscall(callType, uintptr(f.Fd()), 0, 0)
+		if err != nil {
+			log.Fatal(err)
+		}
+		f.Close()
+	}
+}
 
 func main() {
-	syscall.Sync()
+	flag.Parse()
+	switch {
+	case *data:
+		doSyscall(syscall.SYS_FDATASYNC)
```

Comment:
one last little thing? Where possible, it's good to use unix. for system call numbers. But maybe sys_fdatasync is not in unix, it's only in syscall, I don't want to second guess, so I'm approving.

Reply:
> But maybe sys_fdatasync is not in unix

Updated pr to consistently use unix.SYS_ in doSyscall()